### ISSUE TITLE
Error: Distributed operation "GetMNISTData" failed TypeError: Cannot read property 'replace' of undefined

### ISSUE THREAD

**Issue by interactivetech** [deepforge-dev/deepforge#1181]
Hi there,

Really awesome project! I loaded the mnist example and tried to run the pipeline, and I get this error. Can you advise with steps on how to resolve this issue. Please advise if you need additional information!

```
server_1  | 2018-09-10T13:49:32.164Z - error: [gme:server:worker:simpleworker:pid_27:WorkerFunctions] plugin [ExecutePipeline] failed with error Error: Distributed operation "GetMNISTData" failed TypeError: Cannot read property 'replace' of undefined
server_1  |     at finish (/deepforge/node_modules/webgme/src/server/worker/workerrequests.js:130:56)
server_1  |     at ExecutePipeline._callback (/deepforge/node_modules/webgme/src/plugin/managerbase.js:233:21)
server_1  |     at save.then (/deepforge/src/plugins/ExecutePipeline/ExecutePipeline.js:457:30)
server_1  |     at _fulfilled (/deepforge/node_modules/q/q.js:834:54)
server_1  |     at self.promiseDispatch.done (/deepforge/node_modules/q/q.js:863:30)
server_1  |     at Promise.promise.promiseDispatch (/deepforge/node_modules/q/q.js:796:13)
server_1  |     at /deepforge/node_modules/q/q.js:604:44
server_1  |     at runSingle (/deepforge/node_modules/q/q.js:137:13)
server_1  |     at flush (/deepforge/node_modules/q/q.js:125:13)
server_1  |     at _combinedTickCallback (internal/process/next_tick.js:131:7)
server_1  | 2018-09-10T13:49:34.313Z - error: [gme:server:worker:simpleworker:pid_27:WorkerFunctions:PluginManagerBase:plugin:ExecutePipeline] heartbeat failed: Not Found
```

**Comment by brollb** [comment]
Thanks! Have you made any changes to the example or is it the stock mnist example from the examples repo? If so, could you provide the exported branch.

Also, could you provide some information about the deployment? It looks like you are using docker, is that right?

**Comment by interactivetech** [comment]
I used the stock mnist example from the examples repo,  and cloned the master branch of the examples repo from this [link](https://github.com/deepforge-dev/examples)

For deployment, I followed the exact steps in Quickstart on [deepforge.com](http://deepforge.org/)

Here is the error of trying to import the cifar example:
```
server_1  | 2018-09-11T14:23:00.426Z - info: [gme:server:worker:simpleworker:pid_21:WorkerFunctions:PluginManagerBase:plugin:CheckLibraries] Checking for version info at /deepforge/src/seeds/keras/version.txt
server_1  | 2018-09-11T14:23:00.429Z - warn: [gme:server:worker:simpleworker:pid_21:WorkerFunctions:PluginManagerBase:plugin:CheckLibraries] Could not find library version for keras
server_1  | 2018-09-11T14:23:00.431Z - info: [gme:server:worker:simpleworker:pid_21:WorkerFunctions:PluginManagerBase:plugin:CheckLibraries] Checking for version info at /deepforge/src/seeds/pipeline/version.txt
server_1  | 2018-09-11T14:23:00.495Z - warn: [gme:server:worker:simpleworker:pid_21:WorkerFunctions:PluginManagerBase:plugin:CheckLibraries] Could not find library hash for pipeline
server_1  | 2018-09-11T14:23:00.496Z - error: [gme:server:worker:simpleworker:pid_21:WorkerFunctions:PluginManagerBase:plugin:CheckLibraries] Could not check the libraries: TypeError: Cannot read property 'replace' of undefined
server_1  | 2018-09-11T14:23:00.497Z - error: [gme:server:worker:simpleworker:pid_21:WorkerFunctions] plugin [CheckLibraries] failed with error TypeError: Cannot read property 'replace' of undefined
server_1  |     at libs.map.filter.tuple (/deepforge/src/plugins/CheckLibraries/CheckLibraries.js:91:43)
server_1  |     at Array.filter (<anonymous>)
server_1  |     at getAllLibraries.then.libs (/deepforge/src/plugins/CheckLibraries/CheckLibraries.js:89:18)
server_1  |     at _fulfilled (/deepforge/node_modules/q/q.js:834:54)
server_1  |     at self.promiseDispatch.done (/deepforge/node_modules/q/q.js:863:30)
server_1  |     at Promise.promise.promiseDispatch (/deepforge/node_modules/q/q.js:796:13)
server_1  |     at /deepforge/node_modules/q/q.js:604:44
server_1  |     at runSingle (/deepforge/node_modules/q/q.js:137:13)
server_1  |     at flush (/deepforge/node_modules/q/q.js:125:13)
server_1  |     at _combinedTickCallback (internal/process/next_tick.js:131:7)
mongo_1   | 2018-09-11T14:23:24.863+0000 I STORAGE  [conn2] createCollection: deepforge._executorWorkerList with generated UUID: 39a91ea9-0c09-4bd2-b2fe-ec3ada08f025
worker_1  | 2018-09-11T14:23:24.874Z Reconnected to http://server:8888
```

Maybe I need to install deepforge-keras as libraries seem to be missing.

**Comment by brollb** [comment]
Yeah, I think you are right - the examples require the extension but it isn't loaded by default in the docker compose file. I will make an updated one available in the next day or two and update the instructions!

**Comment by brollb** [comment]
Since the examples require the deepforge-keras extension (but this isn't necessarily needed for deepforge itself), I have created another docker-compose file which installs the deepforge-keras extension to the examples repo. The url on deepforge.org has been updated to fetch that docker-compose file instead.

The examples should now work using a deployment spun up with the quickstart! After trying it out, I usually recommend moving to either a native installation or a docker installation (using nvidia-docker for GPU access). It seems that there is support for the nvidia-docker plugin with docker-compose files but I have not experimented with that yet.

Let me know if you have any other issues and thanks for reporting this!
